jujugui/static/gui/src/app/components/user-profile/user-profile.js

```diff
@@ -640,6 +652,66 @@ YUI.add('user-profile', function() {
         </div>);
     },
 
+    /**
+      Display the confirmation for destroying a model.
+
+      @method _displayConfirmation
+      @param {String} model The model to destroy.
+      @return {Object} The confirmation component.
+    */
+    _displayConfirmation: function(model) {
+      this.setState({destroyModel: model});
+    },
+
+    /**
+      Destroy a model.
+
+      @method _destroyModel
+      @param {String} uuid The model uuid to destroy.
+      @return {Object} The confirmation component.
+    */
+    _destroyModel: function(uuid) {
+      this._displayConfirmation(null);
+      this.props.destroyModel((error) => {
+        if (error) {
+          console.log(error);
+          return;
+        }
+        // Reload the models so that the destroyed model updates its status.
+        this._fetchEnvironments(this.props);
+      });
+    },
+
+    /**
+      Generate the confirmation for destroying a model.
+
+      @method _generateConfirmation
+      @return {Object} The confirmation component.
+    */
+    _generateConfirmation: function() {
+      var model = this.state.destroyModel;
+      if (!model) {
+        return;
+      }
+      var buttons = [{
+        title: 'Cancel',
+        action: this._displayConfirmation.bind(this, null),
+        type: 'base'
+      }, {
+        title: 'Destroy',
+        action: this._destroyModel.bind(this, model.uuid),
+        type: 'destructive'
+      }];
+      var name = model.name.split('/')[1];
+      var message = `Are you sure you want to destroy ${name}? This action ` +
```

Comment:
Please also add something similar to "All the services and units included in the model will be destroyed."